packaging_automation/update_docker.py

```diff
@@ -0,0 +1,147 @@
+
+
+def update_changelog(project_version: str, exec_path: str, postgres_version: str = ""):
+    latest_changelog = get_new_changelog_entry(project_version, postgres_version)
+    changelog_file_path = f"{exec_path}/CHANGELOG.md"
+    with open(changelog_file_path, "r+") as reader:
+        if not (f"({project_version}" in reader.readline()):
+            reader.seek(0, 0)
+            old_changelog = reader.read()
+            changelog = f"{latest_changelog}{old_changelog}"
+            reader.seek(0, 0)
+            reader.write(changelog)
+        else:
+            raise ValueError("Already version in the changelog")
+
+
+def update_all_docker_files(project_version: str, tools_path: str, exec_path: str, postgres_version: str):
+    template_path = f"{tools_path}/packaging_automation/templates/docker"
+    pkgvars_file = f"{exec_path}/pkgvars"
+
+    if postgres_version:
+        update_pkgvars(project_version, template_path, pkgvars_file, postgres_version)
+
+    persisted_postgres_version = read_postgres_version(pkgvars_file)
+
+    update_docker_file_for_latest_postgres(project_version, template_path, exec_path, persisted_postgres_version)
+    update_regular_docker_compose_file(project_version, template_path, exec_path)
+    update_docker_file_alpine(project_version, template_path, exec_path, persisted_postgres_version)
+    update_docker_file_for_postgres12(project_version, template_path, exec_path)
+    update_changelog(project_version, exec_path, postgres_version)
+
+
+def read_postgres_version(pkgvars_file: str) -> str:
+    if os.path.exists(pkgvars_file):
+        with open(pkgvars_file, "r") as reader:
+            content = reader.read()
+            lines = content.splitlines()
+            for line in lines:
+                if line.startswith("latest_postgres_version"):
+                    line_parts = line.split("=")
+                    if len(line_parts) != 2:
+                        raise ValueError("keys and values should be seperated with '=' sign")
+                    else:
+                        postgres_version = line_parts[1]
+                else:
+                    raise ValueError("pkgvars file should include a line with key latest_postgres_version")
+    else:
+        # Setting it because pkgvars does not exist initially
+        postgres_version = "13.2"
+    return postgres_version
+
+
+def update_pkgvars(project_version: str, template_path: str, pkgvars_file: str, postgres_version: str):
+    if postgres_version:
+        content = process_template_file(project_version, template_path, "docker-pkgvars.tmpl", postgres_version)
+        with open(pkgvars_file, "w") as writer:
+            writer.write(content)
+
+
+CHECKOUT_DIR = "docker_temp"
+
+
+def create_pr():
+    g = Github(github_token)
+    repository = g.get_repo(f"{REPO_OWNER}/{PROJECT_NAME}")
+    repository.create_pull(title=f"Bump Citus to {args.prj_ver}", base="master",
+                           head=pr_branch, body="")
```

Comment:
I think we should take `pr_branch` & `github_token` as args to our function.
Plus, we can rename `create_pr` to `create_pgxn_pr` or so.

Reply:
Fixed